medicines/api/src/substance.rs

```diff
@@ -1,31 +1,112 @@
 use crate::{pagination, pagination::PageInfo, product::Product};
+use base64;
+use search_client::{models::IndexResults, Search};
+use std::collections::BTreeMap;
 
-#[derive(juniper::GraphQLObject)]
+#[derive(Debug, juniper::GraphQLObject)]
 #[graphql(description = "An active ingredient found in medical products")]
 pub struct Substance {
     name: String,
-    products: Option<Vec<Product>>,
+    products: Vec<Product>,
 }
 
 impl Substance {
-    pub fn new(name: String, products: Option<Vec<Product>>) -> Self {
+    pub fn new(name: String, products: Vec<Product>) -> Self {
         Self { name, products }
     }
-
-    #[allow(dead_code)]
-    fn name(&self) -> &str {
-        &self.name
-    }
 }
 
 pagination! {Substances, SubstanceEdge, Substance}
 
+pub async fn get_substances_starting_with_letter(
+    client: &impl Search,
+    letter: char,
+) -> reqwest::Result<Substances> {
+    let upper_letter = letter.to_ascii_uppercase();
+
+    let azure_result = client
+        .filter_by_field("facets", &upper_letter.to_string())
+        .await?;
+
+    Ok(format_search_results(azure_result, upper_letter))
+}
+
+fn format_search_results(results: IndexResults, letter: char) -> Substances {
+    let mut substances: BTreeMap<&str, BTreeMap<&str, i32>> = BTreeMap::new();
+
+    let letter_string = letter.to_string();
+
+    results
+        .search_results
+        .iter()
+        .filter(|result| result.facets.first() == Some(&letter_string))
```

Comment:
Is it possible to search for index results where facets start with a letter so we don't have to do this?

Reply:
I'm not sure. I haven't checked but I assume not as the current JS code does this as well and I've just been trying to mimic that: https://github.com/MHRA/products/blob/master/medicines/web/src/services/substance-loader.ts#L11

Reply:
Could be a good thing to investigate as an iterative improvement down the line?